stream.go

```diff
@@ -0,0 +1,444 @@
+	RequiredSignedURLs    bool                    `json:"requiredSignedURLs,omitempty"`
+	Watermark             UploadVideoURLWatermark `json:"watermark,omitempty"`
+}
+
+// UploadVideoURLWatermark represents UID of an existing watermark.
+type UploadVideoURLWatermark struct {
+	UID string `json:"uid,omitempty"`
+}
+
+// StreamVideoCreate represents parameters returned after creating a video.
+type StreamVideoCreate struct {
+	UploadURL string               `json:"uploadURL,omitempty"`
+	UID       string               `json:"uid,omitempty"`
+	Watermark StreamVideoWatermark `json:"watermark,omitempty"`
+}
+
+// StreamParameters are the basic parameters needed.
+type StreamParameters struct {
+	AccountID string
+	VideoID   string
+}
+
+// StreamUploadFileParameters are parameters needed for file upload of a video.
+type StreamUploadFileParameters struct {
+	AccountID string
+	VideoID   string
+	FilePath  string
+}
+
+// StreamListParameters represents parameters used when listing stream videos.
+type StreamListParameters struct {
+	AccountID     string
+	VideoID       string
+	After         *time.Time `url:"after,omitempty"`
+	Before        *time.Time `url:"before,omitempty"`
+	Creator       string     `url:"creator,omitempty"`
+	IncludeCounts bool       `url:"include_counts,omitempty"`
+	Search        string     `url:"search,omitempty"`
+	Limit         int        `url:"limit,omitempty"`
+	Asc           bool       `url:"asc,omitempty"`
+	Status        string     `url:"status,omitempty"`
+}
+
+// StreamSignedURLParameters represent parameters used when creating a signed URL.
+type StreamSignedURLParameters struct {
+	AccountID    string
+	VideoID      string
+	ID           string              `json:"id,omitempty"`
+	PEM          string              `json:"pem,omitempty"`
+	EXP          int                 `json:"exp,omitempty"`
+	NBF          int                 `json:"nbf,omitempty"`
+	Downloadable bool                `json:"downloadable,omitempty"`
+	AccessRules  []StreamAccessRules `json:"accessRules,omitempty"`
+}
+
+// StreamVideoResponse represents an API response of a stream video.
+type StreamVideoResponse struct {
+	Response
+	Result StreamVideo `json:"result,omitempty"`
+}
+
+// StreamVideoCreateResponse represents an API response of creating a stream video.
+type StreamVideoCreateResponse struct {
+	Response
+	Result StreamVideoCreate `json:"result,omitempty"`
+}
+
+// StreamListResponse represents the API response from a StreamListRequest.
+type StreamListResponse struct {
+	Response
+	Result []StreamVideo `json:"result,omitempty"`
+	Total  string        `json:"total,omitempty"`
+	Range  string        `json:"range,omitempty"`
+}
+
+// StreamSignedURLResponse represents an API response for a signed URL.
+type StreamSignedURLResponse struct {
+	Response
+	Result struct {
+		Token string `json:"token,omitempty"`
+	}
+}
+
+// StreamAccessRules represents the accessRules when creating a signed URL.
+type StreamAccessRules struct {
```

Comment:
sorry to be a pain but can we rename this (and the usage) to `StreamAccessRule` (it's a single struct we use multiple times).

last one, i promise 😅

Reply:
As long as it is the last one 😆